### ISSUE TITLE
Support late resolution of symbol (reference) information

### ISSUE THREAD

**Issue by akosyakov** [microsoft/language-server-protocol#13]
It would be nice if a location and a container of symbol are resolved later. In our case we have an index of symbols but it is missing information about locations, so to compute them we have to load models for all found symbols. I assume that locations are required for navigation and their can be computed on demand later. The same should be true for references.

**Comment by dbaeumer** [comment]
Good point. This is currently possible for completion items and code actions but not for symbol references.

**Comment by felixfbecker** [comment]
Yes please!

**Comment by yaohaizh** [comment]
Any update on this feature request?  The WorkspaceSymbolProvider.resolveWorkspaceSymbol have this feature but not on the language protocol side.

**Comment by felixfbecker** [comment]
@yaohaizh this is also possible in a push manner instead of pull with #182

**Comment by yaohaizh** [comment]
@felixfbecker, thanks. Will look into the streaming methods.

**Comment by felixfbecker** [comment]
@yaohaizh let me know what you think! you can find a list of language servers that support streaming on http://langserver.org/

**Comment by dbaeumer** [comment]
Implemented for workspace symbols for 3.17.0

**Comment by dbaeumer** [comment]
Reopening, since it is still not there for references.
